src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/EmbeddedLanguages/VirtualChars/CSharpVirtualCharService.cs

```diff
@@ -58,15 +58,19 @@ protected override VirtualCharSequence TryConvertToVirtualCharsWorker(SyntaxToke
 
             if (token.Kind() == SyntaxKind.InterpolatedStringTextToken)
             {
-                // The sections between  `}` and `{` are InterpolatedStringTextToken *as are* the
-                // format specifiers in an interpolated string.  We only want to get the virtual
-                // chars for this first type.
                 if (token.Parent.Parent is InterpolatedStringExpressionSyntax interpolatedString)
                 {
+                    // The sections between  `}` and `{` are InterpolatedStringTextToken *as are* the
+                    // format specifiers in an interpolated string.  We only want to get the virtual
+                    // chars for this first type.
                     return interpolatedString.StringStartToken.Kind() == SyntaxKind.InterpolatedVerbatimStringStartToken
                        ? TryConvertVerbatimStringToVirtualChars(token, "", "", escapeBraces: true)
                        : TryConvertStringToVirtualChars(token, "", "", escapeBraces: true);
                 }
+                else if (token.Parent is InterpolationFormatClauseSyntax)
+                {
+                    return TryConvertStringToVirtualChars(token, "", "", escapeBraces: true);
```

Comment:
are these the actual semantics we want here? i.e. what syntax is allowed inside the format clause.  is it the same as in the rest of the interpolated string?  what if it's a verbatim interpolated string?

Reply:
based on the actual impl, the above is not correct:

```
        private SyntaxToken MakeStringToken(string text, string bodyText, bool isVerbatim, SyntaxKind kind)
        {
            var prefix = isVerbatim ? "@\"" : "\"";
            var fakeString = prefix + bodyText + "\"";
            using (var tempLexer = new Lexer(Text.SourceText.From(fakeString), this.Options, allowPreprocessorDirectives: false))
            {
                LexerMode mode = LexerMode.Syntax;
                SyntaxToken token = tempLexer.Lex(ref mode);
```

Reply:
the format clause follows the same rules wrt verbatim-nes as the rest of the interpolated string.